Wootric/WootricSDK-Android
# Event queue

Add support to handle events. This would bring full support of `Targeted Sampling` to Android. The goal is to be able to call the SDK with an event name and let the server decide if the end user combined with the event name and properties make the user eligible to be surveyed. The implementation adds a concurrent list to handle the events being called in a`FIFO` way.

## Changes
- Add queue to SurveyManager
- Refactor SurveyManager to handle multiple events
- Add call to retrieve event list
- Update and add tests

## Test

- Set up `Targeted Sampling` in one account
- Call survey with the event name and properties to trigger the survey

```java
Wootric wootric = Wootric.init(this, CLIENT_ID, ACCOUNT_TOKEN);
wootric.setEndUserEmail("end_user_email@test.com");


HashMap<String, String> properties = new HashMap();
properties.put("property1", "value1");
properties.put("property2", "value2");

wootric.setProperties(properties);
        
wootric.survey("on_click_event");
```

Try calling this multiple times with different values, the SDK should handle the different calls separately (even try different calls with different `endUserEmail`, `endUserProperties` and `eventName`).

Once the server returns `true`, all subsequent calls should be cancelled and the survey should be shown with the correct values.

- Test this on both smartphones and tablets
